samwise.js

```diff
@@ -1,14 +1,14 @@
 /**
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.